mbtiles/src/mbtiles.rs

```diff
@@ -153,6 +155,62 @@ impl Mbtiles {
         Ok(None)
     }
 
+    /// Get a tile and its hash if it exists from the database
+    ///
+    /// For [`MbtType::Flat`] accessing the hash is not possible, we thus md5 hash the tile data.
+    ///
+    /// See [`Mbtiles::get_tile`] if you don't need the hash
+    pub async fn get_tile_and_hash(
+        &self,
+        conn: &mut SqliteConnection,
+        mbt_type: MbtType,
+        z: u8,
+        x: u32,
+        y: u32,
+    ) -> MbtResult<Option<(Vec<u8>, String)>> {
+        if mbt_type == MbtType::Flat {
+            // we need to register the md5 functions here as the connection might change
+            let mut handle_lock = conn.lock_handle().await?;
```

Comment:
I think we should do this as part of pool creation/mbtiles opening, not on tile retrival.  Moreover, i think this only needs to be done once per file, not once per pool connection

Reply:
That was my initial approach as well.

Thing is, I did not find an option to register functions when a new "thing" in sqlx's Pool gets added. I can change the first "thing", but not subsequent "spans".
Luckyly my tests fail on the naive approach.

We could `load_extension("name")`, but this requires https://github.com/nyurik/sqlite-hashes to be installed on a users device. Not ideal.

Reply:
the reason why I am using the dedicated hashes on the lowest level (instead of the file-level agg hash) is that `ETAG` seem to require that they represent the resource
=> 1/0/1 and 1/0/0 being `ETAG=FOOBAR` would not be allowed.

Thinking about it: I could hash the tile inline without loading the sqlite extension.

Reply:
I think a better path would be to add generic hashing support, and not neseserily the MD5 (which is not that great for many reasons). This way if a tile source has a hash, use that - which would work on all tile sources. If not, compute it on the fly.  I think [xxhash](https://crates.io/crates/xxhash-rust) is the better choice - e.g. the XXH128 is the same size as MD5 IIRC, but much faster to compute.

Reply:
When you say make it generic do you mean:
- please use xxhash here
- change hash to an option, return None if not present => having one level up

Reply:
The later